packages/aws-client/src/lib/S3MultipartUploads.ts

```diff
@@ -0,0 +1,150 @@
+// Utility functions to help with S3 multi-part uploads
+
+import range from 'lodash/range';
+import { s3 } from '../services';
+
+type Chunk = {
+  start: number,
+  end: number
+};
+
+const MB = 1024 * 1024;
+const GB = 1024 * MB;
+
+// Each part of a multi-part copy needs to specify a byte range to be copied.
+// This byte range has a starting byte and an ending byte (inclusive) that makes
+// up the part. The maximum allowed chunk size is 5368709120 bytes.
+//
+// This function takes a file size and an optional maxSize. It returns an array
+// of objects, each containing a `start` and an `end` value. These will make up
+// the ranges of the multi-part copy.
+//
+// From anecdotal testing, a chunk size of 250 MB seems to perform fairly well.
+//
+// https://docs.aws.amazon.com/AmazonS3/latest/API/API_UploadPartCopy.html
+export const createMultipartChunks = (
+  objectSize: number,
+  maxChunkSize = 250 * MB
+): Chunk[] => {
+  if (maxChunkSize > (5 * GB)) {
+    throw new Error('maxChunkSize can be no greater than 5368709120');
+  }
```

Comment:
> Alternatively, I suggest we don't bother checking for either case and simply let AWS throw an error when we pass the "illegal" chunks to uploadPartCopy.

I'm inclined to go with this approach. That way, if AWS adjusts their allowed sizes, we'll get that benefit automatically.

This function is also not part of the public API, it was extracted out to make `multipartCopyObject()` easier to understand. Since it will only be used internally, I'm comfortable with us not passing an invalid value here, or at least noticing when we do.